From 1327e311ca17d1d193aa2e4d3717f7387167d2d5 Mon Sep 17 00:00:00 2001
From: Lei wang <quic_leiwan@quicinc.com>
Date: Wed, 29 Jun 2022 13:27:38 +0800
Subject: [PATCH] lttng-module: fix lttng-module compile issue

Android kernel is a little different from linux standard kernel,
this may cause lttng compile issue.
Make lttng-modules use new kallsyms_lookup_name.
Import android GKI symbol namespace.
Add support for new itimer changes in Android kernel 5.4.

Upstream-Status: Inappropriate [other]

Signed-off-by: Lei wang <quic_leiwan@quicinc.com>
---
 instrumentation/events/lttng-module/timer.h | 2 +-
 wrapper/kallsyms.c                          | 4 ++--
 wrapper/kallsyms.h                          | 3 ++-
 3 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/instrumentation/events/lttng-module/timer.h b/instrumentation/events/lttng-module/timer.h
index 1a45d82..c163656 100644
--- a/instrumentation/events/lttng-module/timer.h
+++ b/instrumentation/events/lttng-module/timer.h
@@ -294,7 +294,7 @@ LTTNG_TRACEPOINT_EVENT_INSTANCE_MAP(timer_hrtimer_class, hrtimer_cancel,
  *		zero, otherwise it is started
  * @expires:	the itimers expiry time
  */
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5,5,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5,4,0))
 LTTNG_TRACEPOINT_EVENT_MAP(itimer_state,
 
 	timer_itimer_state,
diff --git a/wrapper/kallsyms.c b/wrapper/kallsyms.c
index 6af77f5..6c100a4 100644
--- a/wrapper/kallsyms.c
+++ b/wrapper/kallsyms.c
@@ -14,10 +14,10 @@
 #include <linux/module.h>
 #include <wrapper/kallsyms.h>
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5,7,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5,4,0))
 
 #ifndef CONFIG_KPROBES
-# error "LTTng-modules requires CONFIG_KPROBES on kernels >= 5.7.0"
+# error "LTTng-modules requires CONFIG_KPROBES on kernels >= 5.4.0"
 #endif
 
 static
diff --git a/wrapper/kallsyms.h b/wrapper/kallsyms.h
index 5f48847..d54a084 100644
--- a/wrapper/kallsyms.h
+++ b/wrapper/kallsyms.h
@@ -26,8 +26,9 @@
 # endif
 #endif
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5,7,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5,4,0))
 
+MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);
 unsigned long wrapper_kallsyms_lookup_name(const char *name);
 
 #else /* #if (LINUX_VERSION_CODE >= KERNEL_VERSION(5,7,0)) */
-- 
2.35.1

